h2o-algos/src/main/java/hex/deeplearning/DeepLearning.java

```diff
@@ -68,6 +69,25 @@ public ToEigenVec getToEigenVec() {
   @Override public void init(boolean expensive) {
     super.init(expensive);
     _parms.validate(this, expensive);
+    if (expensive) {
+      if (_origTrain != null && _origTrain != _train) {
```

Comment:
unnecessary inner if could be replaced with &&

Reply:
fixed